packages/core/src/domain/configuration/experimentalFeatures.ts

```diff
@@ -25,9 +28,29 @@ export function updateExperimentalFeatures(enabledFeatures: string[] | undefined
 }
 
 export function isExperimentalFeatureEnabled(featureName: string): boolean {
-  return !!enabledExperimentalFeatures && enabledExperimentalFeatures.has(featureName)
+  return (
+    !!enabledExperimentalFeatures &&
+    enabledExperimentalFeatures.has(featureName) &&
+    isExperimentalFeatureSampled(featureName)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Instead of checking if the FF is sampled at every use, you could just:
* set the sample rate before calling `updateExperimentalFeature`
* in `updateExperimentalFeature`, make sure to add only sampled FF to `enableExperimentalFeatures`

Reply:
By doing so the `sampleExperimentalFeature` as to be called before  the `updateExperimentalFeatures` while currently it can be called at any time. Wdyt?

Reply:
Yes, but I don't think we'll ever want to change the sample rate once it's set. I think it's fine.